packages/@aws-cdk/aws-lambda-nodejs/lib/types.ts

```diff
@@ -101,4 +101,53 @@ export interface BundlingOptions {
    * @default - use the Docker image provided by @aws-cdk/aws-lambda-nodejs
    */
   readonly bundlingDockerImage?: BundlingDockerImage;
+
+  /**
+   * Command hooks
+   *
+   * @default - do not run additional commands
+   */
+  readonly commandHooks?: ICommandHooks;
+}
+
+/**
+ * Command hooks
+ *
+ * These commands will run in the environment in which bundling occurs: inside
+ * the container for Docker bundling or on the host OS for local bundling.
+ *
+ * Commands are chained with `&&`.
+ *
+ * @example
+ * {
+ *   // Copy a file from the input directory to the output directory
+ *   // to include it in the bundled asset
+ *   afterBundling(inputDir: string, outputDir: string): string[] {
+ *     return [`cp ${inputDir}/my-binary.node ${outputDir}`];
+ *   }
+ * }
+ */
+export interface ICommandHooks {
+  /**
+   * Returns commands to run before bundling.
+   *
+   * Commands are chained with `&&`.
+   */
+  beforeBundling?(inputDir: string, outputDir: string): string[];
```

Comment:
Does these optional methods work with jsii?

Reply:
It builds but in the `.jsii` I don't see any `optional: true` added to the method's type definition.
Also this pattern is nowhere to be seen in the codebase currently.

Does it mean that it will work for TS but other languages will have to define the 3 methods if `commandHooks` is specified? Can we live with that?

Reply:
@RomainMuller thought?

Reply:
Optional methods are *not* a thing in most languages, so we can't really support that. This is a TypeScript syntax I hadn't seen before... so I guess this is why the compiler does not complain...

Reply:
@eladb should we make the methods mandatory then or leave it like that?

Reply:
Let's make them mandatory so all languages will have a similar experience.